libs/libc/unistd/lib_getopt_common.c

```diff
@@ -394,7 +394,7 @@ int getopt_common(int argc, FAR char * const argv[],
           /* Check for the end of the argument list */
 
           go->go_optptr = argv[go->go_optind];
-          if (!go->go_optptr)
+          if (!go->go_optptr || go->go_optind >= argc)
```

Comment:
maybe better to check `go->go_optind >= argc` before accessing `argv[go->go_optind]` above?

Reply:
@pkarashchenko 
Thanks for your review!
You're right, I force-pushed it.